rusk/tests/common/mod.rs

```diff
@@ -8,32 +8,40 @@ pub mod encoding;
 #[cfg(not(target_os = "windows"))]
 pub mod unix;
 
-use super::SOCKET_PATH;
+use super::{new_socket_path, SOCKET_DIR};
+
+use futures::future::BoxFuture;
+use std::convert::TryFrom;
+use std::path::PathBuf;
+use std::task::{Context, Poll};
+use std::{fs, io};
+
 use futures::TryFutureExt;
 use rusk::services::network::NetworkServer;
 use rusk::services::network::RuskNetwork;
 use rusk::services::pki::KeysServer;
+use rusk::services::prover::ProverServer;
 use rusk::Rusk;
-use std::convert::TryFrom;
 use test_context::AsyncTestContext;
-use tokio::net::UnixListener;
-use tokio::net::UnixStream;
+use tokio::net::{UnixListener, UnixStream};
 use tonic::transport::{Channel, Endpoint, Server, Uri};
-use tower::service_fn;
+use tower::Service;
 use tracing::{subscriber, Level};
 use tracing_subscriber::fmt::Subscriber;
 
 pub struct TestContext {
     pub channel: Channel,
+    socket_path: PathBuf,
 }
 
 #[async_trait::async_trait]
 impl AsyncTestContext for TestContext {
     async fn setup() -> TestContext {
-        // First of all let's remove old stucked socket path
-        // This is required because the tear_down functions is not called if any
-        // test panics
-        let _ = std::fs::remove_file(&*SOCKET_PATH);
+        // Tries to remove the directory - if possible - and recreate it
+        // afterwards. This is required due to the tear_down functions not being
+        // called when a test panics.
+        let _ = fs::remove_dir_all(&*SOCKET_DIR);
```

Comment:
In this case you're "eating" all errors, where we it should be fail for any errors that are not `NotFound`, as shown in the snippet. We should be ok only if the error here is that the folder does not exists, otherwise we should be panic in the same way you do in the teardown.

Reply:
The error should be "eaten". It's on purpose. Even if we can't remove the directory, it shouldn't error out of the function. The reason is because another test might still be busy with a socket.